symengine/series_generic.h

```diff
@@ -8,61 +8,79 @@
 
 #include <list>
 #include <symengine/polynomial.h>
+#include <symengine/rational.h>
+#include <symengine/series.h>
 
 namespace SymEngine {
 //! UnivariateSeries Class
-class UnivariateSeries : public Basic {
-public:
-    //! `var_` : Variable of the UnivariateSeries
-    //! `poly_` : holds the UnivariateIntPolynomial of the series
-    //! `prec_` : precision of the UnivariateSeries, i.e. self = poly + O(var^prec)
+class UnivariateSeries : public SeriesBase<UnivariateExprPolynomial, Expression, UnivariateSeries> {
     // UnivariateSeries 1 + 2*x + x**2 + O(x**5) has dict_ = {{0, 1}, {1, 2}, {2, 1}} with var_ = "x" and prec_ = 5
-    RCP<const Symbol> var_;
-    RCP<const UnivariateIntPolynomial> poly_;
-    unsigned int prec_;
 public:
     IMPLEMENT_TYPEID(UNIVARIATESERIES)
-    //! Constructor of UnivariateSeries class
-    UnivariateSeries(const RCP<const Symbol> &var, const unsigned int &precision, const RCP<const UnivariateIntPolynomial> &poly);
-    UnivariateSeries(const RCP<const Symbol> &var, const unsigned int& precision, const unsigned int& max_exp, map_uint_mpz&& dict);
-    UnivariateSeries(const RCP<const Symbol> &var, const unsigned int &precision, const map_uint_mpz &dict);
-    //! Constructor using a dense vector of integer_class coefficients
+    UnivariateSeries(const UnivariateExprPolynomial &sp, const std::string varname, const unsigned degree) : SeriesBase(std::move(sp), varname, degree) {}
+    UnivariateSeries(const RCP<const Symbol> &var, const unsigned int &precision, const UnivariateExprPolynomial &poly);
+    UnivariateSeries(const RCP<const Symbol> &var, const unsigned int &precision, const unsigned int &max_exp, map_int_Expr &&dict);
+    UnivariateSeries(const RCP<const Symbol> &var, const unsigned int &precision, const map_int_Expr &dict);
+    UnivariateSeries(const RCP<const Symbol> &var, const unsigned int &precision, const std::vector<Expression> &v);
```

Comment:
No, there's already a constructor for a `UnivariatePolynomial` which can be used if needed